subgraphs/saucerswap/v2/src/mappings/position-manager.ts

```diff
@@ -3,36 +3,44 @@ import {
   Collect,
   DecreaseLiquidity,
   IncreaseLiquidity,
-  NonfungiblePositionManager,
-  Transfer
+  NonfungiblePositionManager
 } from '../types/NonfungiblePositionManager/NonfungiblePositionManager'
 import { Bundle, Position, PositionSnapshot, Token } from '../types/schema'
 import { ADDRESS_ZERO, factoryContract, ZERO_BD, ZERO_BI } from '../utils/constants'
-import { Address, BigInt, ethereum } from '@graphprotocol/graph-ts'
+import { Address, BigInt, ethereum, log } from '@graphprotocol/graph-ts'
 import { convertTokenToDecimal, loadTransaction } from '../utils'
 
+const excludedBlocks: BigInt[] = [
+  BigInt.fromI32(56517027),
+  BigInt.fromI32(56520786)
+  // Add more block numbers as needed here
+];
+
+export function isBlockNumberExcluded(blockNumber: BigInt): boolean {
+  return excludedBlocks.includes(blockNumber);
+}
+
 function getPosition(event: ethereum.Event, tokenId: BigInt): Position | null {
+
   let position = Position.load(tokenId.toString())
   if (position === null) {
     let contract = NonfungiblePositionManager.bind(event.address)
     let positionCall = contract.try_positions(tokenId)
 
     // the following call reverts in situations where the position is minted
-    // and deleted in the same block - from my investigation this happens
-    // in calls from  BancorSwap
-    // (e.g. 0xf7867fa19aa65298fadb8d4f72d0daed5e836f3ba01f0b9b9631cdc6c36bed40)
+    // and deleted in the same block
     if (!positionCall.reverted) {
       let positionResult = positionCall.value
-      let poolAddress = factoryContract.getPool(positionResult.value2, positionResult.value3, positionResult.value4)
+      let poolAddress = factoryContract.getPool(positionResult.value0, positionResult.value1, positionResult.value2)
```

Comment:
Seems like there's an offset of 2 here?
ANy context to understand that?

Reply:
Yes, is hard to notice on the ABI, so I am providing with the link to the actual smart contracts of Uniswap v3 and saucerswap v2

Uniswapv3 -->https://github.com/Uniswap/v3-periphery/blob/main/contracts/NonfungiblePositionManager.sol#L85-L96

```
    function positions(uint256 tokenId)
        external
        view
        override
        returns (
            uint96 nonce,
            address operator,
            address token0,
            address token1,
            uint24 fee,
            int24 tickLower,
            int24 tickUpper,
            uint128 liquidity,
            uint256 feeGrowthInside0LastX128,
            uint256 feeGrowthInside1LastX128,
            uint128 tokensOwed0,
            uint128 tokensOwed1
        )
```

Saucerswapv2 --> https://github.com/saucerswaplabs/saucerswaplabs-v2-periphery/blob/master/contracts/NonfungiblePositionManager.sol#L140-L149

```
    function positions(uint256 tokenSN)
        external
        view
        override
        returns (
            address token0,
            address token1,
            uint24 fee,
            int24 tickLower,
            int24 tickUpper,
            uint128 liquidity,
            uint256 feeGrowthInside0LastX128,
            uint256 feeGrowthInside1LastX128,
            uint128 tokensOwed0,
            uint128 tokensOwed1
        )
```

So as you can see, on saucerswap we don't have the first 2 params `nonce` and `operator` hence the results have an offset of 2.